test/e2e/aks_public_ip_prefix.go

```diff
@@ -144,6 +144,27 @@ func AKSPublicIPPrefixSpec(ctx context.Context, inputGetter func() AKSPublicIPPr
 		}, input.WaitIntervals...).Should(Succeed(), "Deleted AzureManagedMachinePool %s/%s still exists", infraMachinePool.Namespace, infraMachinePool.Name)
 	}()
 
+	By("Verifying the AzureManagedMachinePool converges to a failed ready status")
+	Eventually(func(g Gomega) {
+		infraMachinePool := &infrav1.AzureManagedMachinePool{}
+		err := mgmtClient.Get(ctx, client.ObjectKeyFromObject(machinePool), infraMachinePool)
+		g.Expect(err).NotTo(HaveOccurred())
+		cond := conditions.Get(infraMachinePool, infrav1.AgentPoolsReadyCondition)
+		g.Expect(cond).NotTo(BeNil())
+		g.Expect(cond.Status).To(Equal(corev1.ConditionFalse))
+		g.Expect(cond.Reason).To(Equal(infrav1.FailedReason))
+		g.Expect(cond.Message).To(ContainSubstring("PublicIpPrefixOutOfIpAddressesForVMScaleSet"))
```

Comment:
Looks like the error message we used to expect here changed (with SDKv2 I'm assuming).

Reply:
Yup it's a more detailed error message than the previous one, but I think it's the same error.